nibabel/arrayproxy.py

```diff
@@ -45,15 +45,22 @@ class ArrayProxy(object):
     of the numpy dtypes, starting at a given file position ``offset`` with
     single ``slope`` and ``intercept`` scaling to produce output values.
 
-    The class ``__init__`` requires a ``header`` object with methods:
+    The class ``__init__`` can be invoked in two ways:
 
-    * get_data_shape
-    * get_data_dtype
-    * get_data_offset
-    * get_slope_inter
+        (1) ArrayProxy(file_like, header, mmap=True)
 
-    The header should also have a 'copy' method.  This requirement will go away
-    when the deprecated 'header' propoerty goes away.
+            the ``header`` object must have methods:
+
+            * get_data_shape
+            * get_data_dtype
+            * get_data_offset
+            * get_slope_inter
+
+            The header should also have a 'copy' method.  This requirement will
+            go away when the deprecated 'header' property goes away.
+
+        (2) ArrayProxy(file_like, shape, storage_dtype,
```

Comment:
Now I think of it, maybe the second argument can be either:

* the header (as now)
* a tuple of length 2 - 5 containing (shape, storage_dtype, [offset [slope, [inter]]]).

Then the signature can stay the same length, at the expense of slightly greater complexity of explanation / signature.  What do you think?